iquip/apps/builder.py

```diff
@@ -0,0 +1,419 @@
+        """
+        super().__init__(parent=parent)
+        self.experimentPath = experimentPath
+        self.experimentArgs = experimentArgs
+        self.submitted.connect(callback, type=Qt.QueuedConnection)
+
+    def run(self):
+        """Overridden.
+        
+        Submits the experiment to the proxy server.
+
+        If only the experiment was submitted well regardless of whether it runs successfully or not,
+        the server returns the run identifier.
+        After submitted, the submitted signal is emitted.
+        """
+        try:
+            params = {
+                "file": self.experimentPath,
+                "args": json.dumps(self.experimentArgs)
+            }
+        except TypeError:
+            print("Failed to convert the build arguments to a JSON string.")
+        try:
+            response = requests.get("http://127.0.0.1:8000/experiment/submit/",
+                                    params=params,
+                                    timeout=10)
+            response.raise_for_status()
+            rid = response.json()
+        except requests.exceptions.RequestException as err:
+            print(err)
+            return
+        self.submitted.emit(rid)
+
+
+class BuilderApp(qiwis.BaseApp):
+    """App for editting the build arguments and submitting the experiment.
+
+    There are four types of build arguments.
+      BooleanValue: Set to True or False.
+      EnumerationValue: Set to one of the pre-defined candidates.
+      NumberValue: Set to a number with a specific unit and scale.
+      StringValue: Set to a string.
+    
+    Attributes:
+        builderFrame: The frame that shows the build arguments and requests to submit it.
+        experimentPath: The path of the experiment file.
+        experimentClsName: The class name of the experiment.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        experimentPath: str,
+        experimentClsName: str,
+        experimentInfo: Dict[str, Any],
+        parent: Optional[QObject] = None
+    ):  # pylint: disable=too-many-arguments
+        """Extended.
+        
+        Args:
+            experimentPath, experimentClsName: See the attributes section in BuilderApp.
+            experimentInfo: The experiment information, a dictionary of protocols.ExperimentInfo.
+        """
+        super().__init__(name, parent=parent)
+        self.experimentPath = experimentPath
+        self.experimentClsName = experimentClsName
+        self.builderFrame = BuilderFrame()
+        self.initArgsEntry(ExperimentInfo(**experimentInfo))
+        # connect signals to slots
+        self.builderFrame.submitButton.clicked.connect(self.submit)
+
+    def initArgsEntry(self, experimentInfo: ExperimentInfo):
+        """Initializes the build arguments entry.
+        
+        Args:
+            experimentInfo: The experiment information.
+        """
+        for argName, _argInfo in experimentInfo.arginfo.items():
```

Comment:
How about:

```suggestion
        for argName, (argInfo, *_) in experimentInfo.arginfo.items():
```

Reply:
Thanks for your suggestion! I applied it.